server/lib/nl/detection/date.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from datetime import datetime
+
+from server.lib.nl.common.counters import Counters
+from server.lib.nl.detection.types import Date
+from server.lib.nl.detection.types import DateClassificationAttributes
+
+YEAR_RE = [r'(in|after|on|before|year) (\d{4})']
+YEAR_MONTH_RE = [
+    r'(in|after|on|before) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)(?:,)? (\d{4})',
```

Comment:
should we also infer whether it is "in/on" vs. "before" vs. "after" ?

Reply:
Added a preposition field for it